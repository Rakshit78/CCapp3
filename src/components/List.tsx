import {
  TableContainer,
  TableHead,
  Table,
  TableRow,
  TableCell,
  TableBody,
  Pagination,
  CircularProgress,
} from '@mui/material';
import { Box } from '@mui/system';

type Props = {
  page: number;
  setPage: any;
  list: any;
  getjson: any;
  navigate: any;
  loading: boolean;
};

export default function List<Props>({ ...props }): any {
  const scrolltoend = () => {
    props.setPage(props.page + 1);
  };
  window.onscroll = () => {
    if (
      window.scrollY + window.innerHeight >=
      document.documentElement.scrollHeight
    ) {
      scrolltoend();
    }
  };

  return (
    <>
      <form>
        <TableContainer>
          <Table>
            <TableHead>
              <TableRow>
                <TableCell>title</TableCell>
                <TableCell>author</TableCell>
                <TableCell>url</TableCell>
                <TableCell>created at</TableCell>
              </TableRow>
            </TableHead>
            <TableBody>
              {props.list.length > 0 &&
                props.list.map(
                  ({
                    story_title,
                    story_url,
                    author,
                    created_at,
                    objectID,
                  }: any) => {
                    return (
                      <TableRow
                        key={objectID}
                        onClick={() => props.getjson(objectID)}
                      >
                        <TableCell>{story_title}</TableCell>
                        <TableCell>{author}</TableCell>
                        <TableCell>{story_url}</TableCell>
                        <TableCell>{created_at}</TableCell>
                      </TableRow>
                    );
                  }
                )}
            </TableBody>
          </Table>
        </TableContainer>
      </form>
      {props.loading ? (
        <CircularProgress style={{ display: 'grid', placeItems: 'center' }} />
      ) : (
        ''
      )}
    </>
  );
}
