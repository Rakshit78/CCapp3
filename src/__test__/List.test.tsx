import React from 'react';
import { screen, render } from '@testing-library/react';
import List from '../components/List';

describe('render', () => {
  const page: number = 1;
  const list: any = [];
  test('render happen', () => {
    render(
      <List
        setpage={() => {}}
        navigate={() => {}}
        getjsondata={() => {}}
        list={list}
        page={page}
      />
    );
  });

  test('render happen', () => {
    render(
      <List
        setpage={() => {}}
        navigate={() => {}}
        getjsondata={() => {}}
        list={list}
        page={page}
      />
    );
    expect(screen.getByText('created at')).toBeInTheDocument();
  });

  test('render happen', () => {
    render(
      <List
        setpage={() => {}}
        navigate={() => {}}
        getjsondata={() => {}}
        list={list}
        page={page}
      />
    );
    expect(screen.getByText('url')).toBeInTheDocument();
  });

  test('render happen', () => {
    render(
      <List
        setpage={() => {}}
        navigate={() => {}}
        getjsondata={() => {}}
        list={list}
        page={page}
      />
    );
    expect(screen.getByText('title')).toBeInTheDocument();
  });

  test('render happen', () => {
    render(
      <List
        setpage={() => {}}
        navigate={() => {}}
        getjsondata={() => {}}
        list={list}
        page={page}
      />
    );
    expect(screen.getByText('author')).toBeInTheDocument();
  });
});
