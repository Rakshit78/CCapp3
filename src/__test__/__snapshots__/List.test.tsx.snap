// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render render happen 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <div
          class="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
        >
          <table
            class="MuiTable-root css-rqglhn-MuiTable-root"
          >
            <thead
              class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                  scope="col"
                >
                  title
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                  scope="col"
                >
                  author
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                  scope="col"
                >
                  url
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                  scope="col"
                >
                  created at
                </th>
              </tr>
            </thead>
            <tbody
              class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
            />
          </table>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <div
        class="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
      >
        <table
          class="MuiTable-root css-rqglhn-MuiTable-root"
        >
          <thead
            class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                title
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                author
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                url
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                created at
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
          />
        </table>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
